ToDoList-Redux/Binders/TaskListViewModel.swift

```diff
@@ -37,25 +30,24 @@ extension TaskList {
         case add(String)
         case remove(IndexSet)
         case move(IndexSet, Int)
-        case toggle(CheckmarkCellView.Action)
     }
 }
 
 // MARK: - REDUCERS
 
-extension Reducer where ActionType == TaskList.Action, StateType == TaskList.State {
+extension Reducer where ActionType == TaskAction, StateType == [Task] {
```

Comment:
Reducing View State or View Event directly feels a bit wrong. I always prefer isolating Reducers/Middlewares as much as possible from View things. If tomorrow you want to reuse the same logic for an Apple Watch app, where your views have completely different distribution due to the small screen, or if you want to write a console app for example, having the logic tied to View elements will hold you back. Yes, there's an extra step to convert things when you "plug" your views (store projection), but I believe that in this case you could have some weird situations in the future as your app has more screens and share the same model with them. So instead of reducing views, you reduce the model. Views only react to that.

Reply:
So in this case, since this is really a "TaskAction / [Task]" concern, wouldn't it make more sense to move this out of the TaskListViewModel ? It doesn't carry any specificity at this level, imo, and seems like a good candidate to move (back!) to ReduxFramework.swift.

Reply:
Exactly... I didn't move things to make easier the Git Diff, but some things I would put in different folders. Nowadays what I found best for me is organising folders by topics, not by layer. So TaskList would be a folder with TaskListView, TaskListViewModel, TaskListViewProducer and TaskListHandler (that has reducer and middleware). Then another folder TaskDetails...